test/jdk/sun/util/calendar/CalendarSystemDeadLockTest.java

```diff
@@ -0,0 +1,171 @@
+        }
+        // intentional identity check since sun.util.calendar.CalendarSystem#getGregorianCalendar() is
+        // expected to return a singleton instance
+        if ((gCal != GetGregorianCalTask.instances.get(0)) || (gCal != GetGregorianCalTask.instances.get(1))) {
+            throw new RuntimeException("sun.util.calendar.CalendarSystem#getGregorianCalendar()" +
+                    " returned different instances");
+        }
+    }
+
+    /**
+     * Reflectively calls sun.util.calendar.CalendarSystem#getGregorianCalendar() and returns
+     * the result
+     */
+    private static Object callCalSystemGetGregorianCal() throws Exception {
+        final Class<?> k = Class.forName("sun.util.calendar.CalendarSystem");
+        return k.getDeclaredMethod("getGregorianCalendar").invoke(null);
+    }
+
+    private static class ClassLoadTask implements Callable<Class<?>> {
+        private final String className;
+        private final CountDownLatch latch;
+
+        private ClassLoadTask(final String className, final CountDownLatch latch) {
+            this.className = className;
+            this.latch = latch;
+        }
+
+        @Override
+        public Class<?> call() {
+            System.out.println(Thread.currentThread().getName() + " loading " + this.className);
+            try {
+                // let the other tasks know we are ready to trigger our work
+                latch.countDown();
+                // wait for the other task to let us know they are ready to trigger their work too
+                latch.await();
+                return Class.forName(this.className);
+            } catch (Exception e) {
+                throw new RuntimeException(e);
+            }
+        }
+    }
+
+    private static class GetGregorianCalTask implements Callable<Object> {
+        // keeps track of the instances returned by calls to sun.util.calendar.CalendarSystem#getGregorianCalendar()
+        // by this task
+        private static final List<Object> instances = Collections.synchronizedList(new ArrayList<>());
+        private final CountDownLatch latch;
+
+        private GetGregorianCalTask(final CountDownLatch latch) {
+            this.latch = latch;
+        }
+
+        @Override
+        public Object call() {
+            System.out.println(Thread.currentThread().getName()
+                    + " calling  sun.util.calendar.CalendarSystem#getGregorianCalendar()");
+            try {
+                // let the other tasks know we are ready to trigger our work
+                latch.countDown();
+                // wait for the other task to let us know they are ready to trigger their work too
+                latch.await();
+                final Object inst = callCalSystemGetGregorianCal();
+                instances.add(inst);
+                return inst;
+            } catch (Exception e) {
+                throw new RuntimeException(e);
+            }
+        }
+    }
+}
```

Comment:
Need a new line at the EOF.

Reply:
Done. I've updated this in the latest version of the PR.